src/main/java/org/kainos/ea/db/AuthDao.java

```diff
@@ -0,0 +1,52 @@
+package org.kainos.ea.db;
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import org.kainos.ea.cli.LoginRequest;
+import org.kainos.ea.cli.Role;
+import org.kainos.ea.cli.User;
+import org.kainos.ea.client.FailedLoginException;
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.security.Key;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Date;
+
+
+public class AuthDao {
+    private  DatabaseConnector databaseConnector;
+
+    public AuthDao(DatabaseConnector databaseConnector) {
+        this.databaseConnector=databaseConnector;
+    }
+
+
+    public User validLogin(LoginRequest login) throws SQLException, FailedLoginException {
+        Connection c = databaseConnector.getConnection();
+
+        PreparedStatement ps = c.prepareStatement("SELECT UserID, Email, Password, RoleID FROM `Users` WHERE Email='"
+                + login.getEmail() + "'");
+
+        ResultSet rs = ps.executeQuery();
+
+        if (rs.next()) {
+            String hashedPassword = rs.getString("Password");
```

Comment:
This function breaks the SRP and has two functions:

1) Get data from the database
2) Validate the password from the database